### PR TITLE
Fix bugs with translations

### PR DESCRIPTION
**Pull request by Pavl0-Gural**
Fix 500 error when try to change language from eng to ua

### CODE REVIEW

**Review by SomeOne35719 — APPROVED** [review]

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 1]
```diff
@@ -503,7 +508,9 @@ private AddressInfoDto addressInfoDtoBuilder(Order order) {
             BagForUserDto bagDto = new BagForUserDto();
             bagDto.setCount(amountOfBags.get(bag.getId()));
             List<BagTranslation> translations = bag.getBagTranslations();
-            if (languageId == 2) {
+            if (translations.size() == 1) {
+                bagDto.setService(translations.get(0).getName());
+            } else if (languageId == 2) {
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: why you don't use the binary operator OR ( || )
 if(translations.size() == 1) || languageId == 2) ?

File: core/src/main/java/greencity/controller/ClientController.java [hunk 2]
```diff
@@ -65,7 +66,8 @@ public ClientController(UBSClientService ubsClientService) {
         @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
     })
     @GetMapping("/get-all-orders-data/{lang}")
-    public ResponseEntity<List<OrderStatusForUserDto>> getAllDataForOrder(
+    @ApiPageable
+    public ResponseEntity<PageableDto<OrderStatusForUserDto>> getAllDataForOrder(
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/controller/ClientController.java:70]: Please check https://restfulapi.net/

**Review by VadymMakitra — COMMENTED** [review]
Please resolve comment's

**Comment by VadymMakitra** [comment]
image
You pr have only 36 % of coverage. Please up the coverage of you pr at least to 70 %

**Review by VadymMakitra — CHANGES_REQUESTED** [review]
Please resolve comments

**Comment by VadymMakitra** [comment]
@Pavl0-Gural Please resolve previous comment's

File: core/src/main/java/greencity/controller/ClientController.java [hunk 3]
```diff
@@ -64,10 +65,11 @@ public ResponseEntity<List<OrderClientDto>> getAllOrdersDoneByUser(
         @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
         @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
     })
-    @GetMapping("/get-all-orders-data/{lang}")
-    public ResponseEntity<List<OrderStatusForUserDto>> getAllDataForOrder(
-        @ApiIgnore @CurrentUserUuid String uuid, @PathVariable Long lang, @ApiIgnore Pageable page) {
-        return ResponseEntity.status(HttpStatus.OK).body(ubsClientService.getOrdersForUser(uuid, lang, page));
+    @GetMapping("/get-all-orders-data")
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/controller/ClientController.java]: please change the web address to the rest API standard. Additional info here: [REST-API](https://restfulapi.net/)

**Review by VadymMakitra — COMMENTED** [review]
Please resolve comment;

**Review by VadymMakitra — APPROVED** [review]

**Review by Andrii-Yezenitskyi — APPROVED** [review]

File: core/src/main/java/greencity/controller/ClientController.java [hunk 4]
```diff
@@ -64,10 +65,11 @@ public ResponseEntity<List<OrderClientDto>> getAllOrdersDoneByUser(
         @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
         @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
     })
-    @GetMapping("/get-all-orders-data/{lang}")
-    public ResponseEntity<List<OrderStatusForUserDto>> getAllDataForOrder(
-        @ApiIgnore @CurrentUserUuid String uuid, @PathVariable Long lang, @ApiIgnore Pageable page) {
-        return ResponseEntity.status(HttpStatus.OK).body(ubsClientService.getOrdersForUser(uuid, lang, page));
+    @GetMapping("/get-user-orders")
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/ClientController.java]: avoid using "get" in URL name. read this: https://restfulapi.net/resource-naming/

File: core/src/main/java/greencity/controller/ClientController.java [hunk 5]
```diff
@@ -64,10 +65,11 @@ public ResponseEntity<List<OrderClientDto>> getAllOrdersDoneByUser(
         @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
         @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/ClientController.java]: This API never retirns 404, please keep doc up to date.

File: dao/src/main/resources/db/changelog/logs/ch-update-order_payment_status_translations-Hural.xml [hunk 6]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet id="Hural-3" author="Pavlo Hural">
```
Comment (thesvider) [inline dao/src/main/resources/db/changelog/logs/ch-update-order_payment_status_translations-Hural.xml]: no need to have 2 separate changesets Hural-3 and Hural-4. You can merge them into single one

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 7]
```diff
@@ -430,48 +431,70 @@ public MakeOrderAgainDto makeOrderAgain(Locale locale, Long orderId) {
...
+    @Override
+    public PageableDto<OrdersDataForUserDto> getOrdersForUser(String uuid, Pageable page) {
         PageRequest pageRequest = PageRequest.of(page.getPageNumber(), page.getPageSize());
         Page<Order> orderPages = ordersForUserRepository.findAllOrdersByUserUuid(pageRequest, uuid);
         List<Order> orders = orderPages.getContent();
 
-        List<OrderStatusForUserDto> dtos = new ArrayList<>();
+        List<OrdersDataForUserDto> dtos = new ArrayList<>();
 
         for (Order order : orders) {
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Simplify this loop. You can use stream + introduce private methods

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 8]
```diff
@@ -430,48 +431,70 @@ public MakeOrderAgainDto makeOrderAgain(Locale locale, Long orderId) {
...
+        }
+
+        return new PageableDto<>(
+            dtos,
+            orderPages.getTotalElements(),
+            orderPages.getPageable().getPageNumber(),
+            orderPages.getTotalPages());
+    }
+
+    private List<CertificateDto> getCertificate(Set<Certificate> certificates) {
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: this logic should be in mapper.

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 9]
```diff
@@ -430,48 +431,70 @@ public MakeOrderAgainDto makeOrderAgain(Locale locale, Long orderId) {
...
+        List<CertificateDto> dto = new ArrayList<>();
+        for (Certificate certificate : certificates) {
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: `    private List<CertificateDto> getCertificate(Set<Certificate> certificates) {
        return certificates.stream()
                .map(certificate -> CertificateDto.builder()
                        .certificateDate(certificate.getCreationDate())
                        .certificateStatus(certificate.getCertificateStatus().name())
                        .certificatePoints(certificate.getPoints())
                        .code(certificate.getCode())
                        .build())
                .collect(Collectors.toList())
    }`

File: service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java [hunk 10]
```diff
@@ -379,58 +376,63 @@ private AddressExportDetailsDto getAddressDtoForAdminPage(Address address) {
...
         Optional<OrderStatusTranslation> orderStatusTranslation =
-            orderStatusTranslationRepository.getOrderStatusTranslationByIdAndLanguageId(orderStatus.getNumValue(),
-                languageRepository.findIdByCode(language.getCode()));
+            orderStatusTranslationRepository.getOrderStatusTranslationById(orderStatus.getNumValue());
         String currentOrderStatusTranslation =
             orderStatusTranslation.isPresent() ? orderStatusTranslation.get().getName() : "order status not found";
-        String currentOrderStatusPaymentTranslation = null;
+        String currentOrderStatusTranslationEng =
+            orderStatusTranslation.isPresent() ? orderStatusTranslation.get().getNameEng()
+                : "order status english not found";
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java]: this is wrong. I'd better return enum value (OrderStatus) than this message. It's confusing for user

File: service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java [hunk 11]
```diff
@@ -379,58 +376,63 @@ private AddressExportDetailsDto getAddressDtoForAdminPage(Address address) {
...
+
+        OrderPaymentStatusTranslation currentOrderStatusPaymentTranslation = new OrderPaymentStatusTranslation();
+        Order currentOrder = new Order();
         if (order.isPresent()) {
+            currentOrder = order.get();
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java]: it's better to use Optional streams instead fof these condition on isPresent

**Review by thesvider — CHANGES_REQUESTED** [review]
please resolve comments

**Review by thesvider — APPROVED** [review]

**Pull request merged**